src/main/java/net/floodlightcontroller/perfmon/CumulativeTimeBucketJSONSerializer.java

```diff
@@ -36,7 +36,7 @@ public void serialize(CumulativeTimeBucket ctb,
                    SerializerProvider serializer) 
                    throws IOException, JsonProcessingException {
        jGen.writeStartObject();
-       Timestamp ts = new Timestamp(ctb.getStartTimeNs()/1000000);
+       Timestamp ts = new Timestamp(System.currentTimeMillis() - (System.nanoTime() - ctb.getStartTimeNs())/1000000);
        jGen.writeStringField("start-time", ts.toString());
```

Comment:
I’d recommend getting the time once via nanoTime() and save it to a variable. Then, convert that to milliseconds where you need it. This removes one syscall and also eliminates variability in time stamps

Reply:
Thanks for the suggestion, will do.